system-helper/flatpak-system-helper.c

```diff
@@ -46,11 +46,63 @@ static gboolean no_idle_exit = FALSE;
 typedef PolkitAuthorizationResult AutoPolkitAuthorizationResult;
 typedef PolkitDetails             AutoPolkitDetails;
 typedef PolkitSubject             AutoPolkitSubject;
+typedef struct _FlatpakSystemHelperOngoingPull FlatpakSystemHelperOngoingPull;
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (AutoPolkitAuthorizationResult, g_object_unref)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (AutoPolkitDetails, g_object_unref)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (AutoPolkitSubject, g_object_unref)
 
+struct _FlatpakSystemHelperOngoingPull
+{
+  FlatpakSystemHelper *object;
+  GDBusMethodInvocation *invocation;
+
+  const gchar *commit_id;
+  const gchar *ref;
+  guint watch_id;
+  uid_t uid;
+  guint64 current_timestamp;
+
+  gchar *root_folder_name;
+  gchar *user_folder_name;
+
+  GUnixMountMonitor *mount_monitor;
+  gulong mount_monitor_id;
+  GSubprocess *bindfs;
+};
+
+static void
+system_helper_ongoing_pull_free (FlatpakSystemHelperOngoingPull *pull)
```

Comment:
A bit surprising to see a _free  function that doesn't actually free its argument.